search/services/index.py

```diff
@@ -0,0 +1,785 @@
+                    )
+                    pass
+            result[key] = value
+        result['score'] = raw.meta.score
+        return Document(**result)   # type: ignore
+        # See https://github.com/python/mypy/issues/3937
+
+
+def init_app(app: object = None) -> None:
+    """Set default configuration parameters for an application instance."""
+    config = get_application_config(app)
+    config.setdefault('ELASTICSEARCH_HOST', 'localhost')
+    config.setdefault('ELASTICSEARCH_PORT', '9200')
+    config.setdefault('ELASTICSEARCH_INDEX', 'arxiv')
+    config.setdefault('ELASTICSEARCH_USER', None)
+    config.setdefault('ELASTICSEARCH_PASSWORD', None)
+    config.setdefault('ELASTICSEARCH_MAPPING', 'mappings/DocumentMapping.json')
+
+
+# TODO: consider making this private.
+def get_session(app: object = None) -> SearchSession:
+    """Get a new session with the search index."""
+    config = get_application_config(app)
+    host = config.get('ELASTICSEARCH_HOST', 'localhost')
+    port = config.get('ELASTICSEARCH_PORT', '9200')
+    scheme = config.get('ELASTICSEARCH_SCHEME', 'http')
+    index = config.get('ELASTICSEARCH_INDEX', 'arxiv')
+    user = config.get('ELASTICSEARCH_USER', None)
+    password = config.get('ELASTICSEARCH_PASSWORD', None)
+    mapping = config.get('ELASTICSEARCH_MAPPING',
+                         'mappings/DocumentMapping.json')
+    return SearchSession(host, index, port, scheme, user, password, mapping)
+
+
+# TODO: consider making this private.
+def current_session() -> SearchSession:
+    """Get/create :class:`.SearchSession` for this context."""
+    g = get_application_global()
+    if not g:
+        return get_session()
+    if 'search' not in g:
+        g.search = get_session()    #type: ignore
+    return g.search     # type: ignore
+
+
+@wraps(SearchSession.search)
+def search(query: Query) -> DocumentSet:
+    """Retrieve search results."""
+    return current_session().search(query)
+
+
+@wraps(SearchSession.add_document)
+def add_document(document: Document) -> None:
+    """Add Document."""
+    return current_session().add_document(document)
+
+
+@wraps(SearchSession.bulk_add_documents)
+def bulk_add_documents(documents: List[Document]) -> None:
+    """Add Documents."""
+    return current_session().bulk_add_documents(documents)
+
+
+@wraps(SearchSession.get_document)
+def get_document(document_id: int) -> Document:
+    """Retrieve arxiv document by id."""
+    return current_session().get_document(document_id)
+
+
+@wraps(SearchSession.cluster_available)
+def cluster_available() -> bool:
+    """Check whether the cluster is available."""
+    return current_session().cluster_available()
+
+
+@wraps(SearchSession.create_index)
+def create_index() -> None:
+    """Create the search index."""
+    current_session().create_index()
+
+
+def ok() -> bool:
+    """Health check."""
+    try:
+        current_session()
+    except Exception:    # TODO: be more specific.
```

Comment:
Yes, please be more specific.  This function has unclear context and does not appear to bubble up more informative errors.

Reply:
Indeed. This should be addressed in the context of #104